### ISSUE TITLE
Preloaded values cleared when using x-editable with select2

### ISSUE THREAD

**Issue by AnalogJ** [vitalets/x-editable#316]
I have some preloaded values for a select2 input that are specified as follows: `data-value="12345"`

Once I attempt to add a new value using a filtered ajax query for select2, the previously preloaded values are wiped out. I think this may be due to this line in the source code.

[https://github.com/vitalets/x-editable/blob/master/src/inputs/select2/select2.js#L102](https://github.com/vitalets/x-editable/blob/master/src/inputs/select2/select2.js#L102)

```
//when data is loaded via ajax, we need to know when it's done to populate listData
        if(this.isRemote) {
            //listen to loaded event to populate data
            this.$input.on('select2-loaded', $.proxy(function(e) {
                this.sourceData = e.items.results;
            }, this));
        }
```

Basically what's happening is that the sourceData is wiped out and replaced. This means when the $.fn.editableutils.itemsByValue() function is called, the sourceData no longer has the preloaded tag id's are they are removed when the new html is generated.
